Mousewheel click opens new tab at selected path

Raithlin opened the pull request:
<!-- 
🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨
I ACKNOWLEDGE THE FOLLOWING BEFORE PROCEEDING:
1. PR may be deleted if it is not following the template
2. Try not to make duplicates. Do a quick search before posting
3. Add a clarified title
-->

**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #3667

**Details of Changes**
Add details of changes here.
- Added method to capture mouse clicks. Middle clicks are filtered and a new tab opened.


**Validation**
How did you test these changes?
- [X ] Built and ran the app
- [X ] Tested the changes for accessibility

**Screenshots (optional)**
Add screenshots here.

yaichenbaum commented on Files.Launcher/Win32API.cs:
```diff
@@ -21,553 +21,553 @@
+		public static extern int SHQueryRecycleBin(string pszRootPath,
+			ref SHQUERYRBINFO pSHQueryRBInfo);
+	}
 }
```
Can you fix the spacing?

yaichenbaum commented on Files.Launcher/Win32API.cs:
```diff
@@ -21,553 +21,553 @@
+		public static extern int SHQueryRecycleBin(string pszRootPath,
+			ref SHQUERYRBINFO pSHQueryRBInfo);
+	}
 }
```
Can you fix the tab spacing?

Raithlin replied:
I reverted the change to Win32Api entirely. It is not a part of this commit.

BanCrash commented:
@Raithlin do you know why sometimes it opens the same folder in both new tab and actual tab? I can't replicate consistently, but it happens some times in my tests without any apparently reason.

![GIF 02-12-2021 14-41-20](https://user-images.githubusercontent.com/11770760/144433266-b91c5a20-f4f4-411a-bfda-ddcc3c64ea17.gif)

Raithlin commented:
I had that happen once, thought I had clicked on something else. I can take a look - most likely the click method is being called as well. I can't check the mouse status in that method either, so it makes it a little more difficult to prevent. Any ideas?

Raithlin commented:
I added logging (temporarily) to see which methods were being called. Both were. 
I changed the process to use a private pointerroutereventargs to pass to the tapped event as per https://stackoverflow.com/questions/44111224/pointerpressed-vs-tapped#44125921. No click through happening any more, and the middle click still works.

BanCrash commented on Files/ViewModels/NavToolbarViewModel.cs:
```diff
@@ -604,9 +606,36 @@ public void VisiblePath_QuerySubmitted(AutoSuggestBox sender, AutoSuggestBoxQuer
         }
 
-        public void PathBoxItem_Tapped(object sender, TappedRoutedEventArgs e)
+        public async Task PathBoxItem_PointerPressed(object sender, PointerRoutedEventArgs e)
```
Any reason for this to be an async Task method? IMO a public void would be enough.

Raithlin replied:
I was following best practice (and Roslyn analyser recommendations) - but you are right. It isn't being called by an async method, so either way it doesn't make any difference.

BanCrash replied:
Ok, then:

```suggestion
        public void PathBoxItem_PointerPressed(object sender, PointerRoutedEventArgs e)
```

BanCrash reviewed (approved):
LGTM
